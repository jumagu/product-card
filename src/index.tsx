export * from './hooks';
export * from './components';
export * from './interfaces/product.interfaces';
