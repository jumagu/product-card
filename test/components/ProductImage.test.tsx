import React from 'react';
import renderer from 'react-test-renderer';

import { product2 } from '../data/products';
import { ProductImage, ProductCard } from '../../src/components';

describe('ProductImage', () => {
  test('Should render correctly with the custom image', () => {
    const wrapper = renderer.create(
      <ProductImage img='test-product.jpg'></ProductImage>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Should render correctly with the product image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {() => <ProductImage></ProductImage>}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
