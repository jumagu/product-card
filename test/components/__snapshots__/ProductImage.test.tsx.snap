// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Should render correctly with the custom image 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="test-product.jpg"
/>
`;

exports[`ProductImage Should render correctly with the product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./t-shirt.png"
  />
</div>
`;
